### PR TITLE
Support same dispatcher as parent in Typed, #27123

### PR DESCRIPTION
**Pull request by patriknw**
* Adding `DispatcherSelector.sameAsParent`

Refs #27123

### CODE REVIEW

File: akka-actor-typed/src/main/scala/akka/actor/typed/internal/PropsImpl.scala [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    override def withNext(next: Props): Props = copy(next = next)
+  }
+  object DispatcherDefault {
+    // this is hidden in order to avoid match on this object
+    private val empty = DispatcherDefault(EmptyProps)
+
+    /**
+     * Retrieve an instance for this configuration node with empty `next` reference.
+     */
+    def apply(): DispatcherDefault = empty
```
Comment (johanandren) [inline akka-actor-typed/src/main/scala/akka/actor/typed/internal/PropsImpl.scala]: Is there still a reason to keep the `apply` now that the entire thing is internal?

**Review by johanandren — APPROVED** [review]

File: akka-actor-typed/src/main/scala/akka/actor/typed/internal/PropsImpl.scala (hunk 1 above)
Comment (patriknw) [reply akka-actor-typed/src/main/scala/akka/actor/typed/internal/PropsImpl.scala]: ok, removed in fed7099

File: akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/DispatcherSelectorSpec.scala [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+      val probe = createTestProbe[Pong]()
+      val child = childProbe.receiveMessage()
+      child ! Ping(probe.ref)
+
+      val response = probe.receiveMessage()
+      response.threadName should startWith("DispatcherSelectorSpec-ping-pong-dispatcher")
+    }
+
+    "use default dispatcher if selecting parent dispatcher for user guardian" in {
```
Comment (jrudolph) [inline akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/DispatcherSelectorSpec.scala]: Sounds like a sensible default.

File: akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/DispatcherSelectorSpec.scala [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+      val probe = createTestProbe[Pong]()
+      val child = childProbe.receiveMessage()
+      child ! Ping(probe.ref)
+
+      val response = probe.receiveMessage()
+      response.threadName should startWith("DispatcherSelectorSpec-ping-pong-dispatcher")
+    }
+
+    "select same dispatcher as parent, several levels" in {
```
Comment (jrudolph) [inline akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/DispatcherSelectorSpec.scala]: :+1: good to check nesting.

File: akka-actor/src/main/scala/akka/actor/Deployer.scala [hunk 4]
```diff
@@ -10,13 +10,19 @@ import akka.routing._
...
+
 object Deploy {
   final val NoDispatcherGiven = ""
   final val NoMailboxGiven = ""
   val local = Deploy(scope = LocalScope)
+
+  /**
+   * INTERNAL API
+   */
+  @InternalApi private[akka] final val DispatcherSameAsParent = ".."
```
Comment (jrudolph) [inline akka-actor/src/main/scala/akka/actor/Deployer.scala]: :)

**Review by jrudolph — APPROVED** [review]
LGTM

**Pull request merged**